rdr_service/dao/genomics_dao.py

```diff
@@ -411,6 +412,60 @@ def get_null_field_members(self, field):
             ).all()
         return members
 
+    def get_unconsented_gror_since_date(self, _date):
+        """
+        Get the genomic set members with GROR updated to No Consent since date
+        :param _date:
+        :return: GenomicSetMember list
+        """
+        with self.session() as session:
+            members = session.query(GenomicSetMember).join(
+                (ParticipantSummary,
+                 GenomicSetMember.participantId == ParticipantSummary.participantId)
+            ).filter(
+                GenomicSetMember.genomicWorkflowState.in_((
+                    GenomicWorkflowState.GEM_RPT_READY,
+                    GenomicWorkflowState.A1,
+                    GenomicWorkflowState.A2
+                )) &
+                (
+                    (ParticipantSummary.consentForGenomicsROR != QuestionnaireStatus.SUBMITTED) |
+                    (ParticipantSummary.consentForStudyEnrollment != QuestionnaireStatus.SUBMITTED)
+                ) &
+                (
+                    (ParticipantSummary.consentForGenomicsRORAuthored > _date) |
+                    (ParticipantSummary.consentForStudyEnrollmentAuthored > _date)
+                )
```

Comment:
Should there be a logical link between checking the status and authored time for that status? This seems like it could find a participant that has a no consent status for GROR authored before the date, but has made a change to their 'study for enrollment' status after the date.

Same sort of concept down in the reconsented method.

Reply:
Yes, I see what you mean. It should be (pseudo-code) `(ror_consent != 1 AND ror_authored > date) OR (general_consent != 1 AND consent_authored > date)`.

Reply:
Looks good 👍